// react-native/Libraries/Animated/Animated.js
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 * @format
 */

'use strict';

import Platform from 'react-native/Libraries/Utilities/Platform';
import typeof AnimatedFlatList from './components/AnimatedFlatList';
import typeof AnimatedImage from './components/AnimatedImage';
import typeof AnimatedScrollView from './components/AnimatedScrollView';
import typeof AnimatedSectionList from './components/AnimatedSectionList';
import typeof AnimatedText from './components/AnimatedText';
import typeof AnimatedView from './components/AnimatedView';

const AnimatedMock = require('react-native/Libraries/Animated/AnimatedMock');
const AnimatedImplementation = require('react-native/Libraries/Animated/AnimatedImplementation');

const Animated = ((Platform.isTesting
  ? AnimatedMock
  : AnimatedImplementation): typeof AnimatedMock);

module.exports = {
  get FlatList(): AnimatedFlatList {
    return require('react-native/Libraries/Animated/components/AnimatedFlatList');
  },
  get Image(): AnimatedImage {
    return require('react-native/Libraries/Animated/components/AnimatedImage');
  },
  get ScrollView(): AnimatedScrollView {
    return require('react-native/Libraries/Animated/components/AnimatedScrollView');
  },
  get SectionList(): AnimatedSectionList {
    return require('react-native/Libraries/Animated/components/AnimatedSectionList');
  },
  get Text(): AnimatedText {
    return require('react-native/Libraries/Animated/components/AnimatedText');
  },
  get View(): AnimatedView {
    return require('react-native/Libraries/Animated/components/AnimatedView');
  },
  ...Animated,
};
